assets/quick-order-list.js

```diff
@@ -208,12 +212,38 @@ class QuickOrderList extends HTMLElement {
     this.defineInputsAndQuickOrderTable();
   }
 
+  scrollQuickOrderListTable() {
+    const inputTopBorder = this.variantListInput.getBoundingClientRect().top;
+    const inputBottomBorder = this.variantListInput.getBoundingClientRect().bottom;
+
+    if (this.isListInsideModal) {
+      const totalBarCrossesInput = inputBottomBorder > this.totalBar.getBoundingClientRect().top;
+      const tableHeadCrossesInput = inputTopBorder < this.tableHead.getBoundingClientRect().bottom;
+
+      if (totalBarCrossesInput || tableHeadCrossesInput) {
+        this.variantListInput.scrollIntoView({ block: 'center', behavior: 'smooth' });
+      }
+    } else {
+      const stickyHeaderBottomBorder = this.stickyHeaderElement && this.stickyHeaderElement.getBoundingClientRect().bottom;
+      const totalBarCrossesInput = inputBottomBorder > this.totalBarPosition;
+      const inputOutsideOfViewPort = inputBottomBorder < this.inputFieldHeight;
+      const stickyHeaderCrossesInput = this.stickyHeaderElement && this.stickyHeader.type !== 'on-scroll-up' && this.stickyHeader.height > inputTopBorder;
+      const stickyHeaderScrollupCrossesInput = this.stickyHeaderElement && this.stickyHeader.type === 'on-scroll-up' && this.stickyHeader.height > inputTopBorder && stickyHeaderBottomBorder > 0;
+
+      if (totalBarCrossesInput || inputOutsideOfViewPort || stickyHeaderCrossesInput || stickyHeaderScrollupCrossesInput) {
+        this.variantListInput.scrollIntoView({ block: 'center', behavior: 'smooth' });
```

Comment:
Just an idea but if this and what youre doing in line 224 are the same we can always make it a function to avoid duplicating the same line

And just call the function in the two places `this.scrollIntoView()`

```
scrollIntoView() {
 this.variantListInput.scrollIntoView({ block: 'center', behavior: 'smooth' });
}
```

Thoughts?

Reply:
Love this idea even though there are just two repetitive lines, still looks better. Thanks for the idea @sofiamatulis